cvmfs/network/download.cc

```diff
@@ -1676,6 +1693,21 @@ Failures DownloadManager::Fetch(JobInfo *info) {
     info->info_header()[header_size-1] = '\0';
   }
 
+  if (enable_http_tracing_) {
+    const std::string str_pid = "X-CVMFS-PID: " + StringifyInt(info->pid());
+    const std::string str_gid = "X-CVMFS-GID: " + StringifyUint(info->gid());
+    const std::string str_uid = "X-CVMFS-UID: " + StringifyUint(info->uid());
+
+    // will be auto freed at the end of this function Fetch(JobInfo *info)
+    info->SetTracingHeaderPid(static_cast<char *>(alloca(str_pid.size() + 1)));
+    info->SetTracingHeaderGid(static_cast<char *>(alloca(str_gid.size() + 1)));
+    info->SetTracingHeaderUid(static_cast<char *>(alloca(str_uid.size() + 1)));
+
+    memcpy(info->tracing_header_pid(), str_pid.c_str(), str_pid.size() + 1);
+    memcpy(info->tracing_header_gid(), str_gid.c_str(), str_gid.size() + 1);
+    memcpy(info->tracing_header_uid(), str_uid.c_str(), str_uid.size() + 1);
```

Comment:
Since we are anyway using (short-lived) `std::string` instances here, there is not much point in using `alloca` in addition.  I think we can either go for `std::string::c_str()` or we'd use `alloca()` and C string manipulation, which should be a bit faster but may not be worth the effort.

Reply:
`alloca` we decided to use because of the scope of the variables life